### ISSUE TITLE
Graphs

### ISSUE THREAD

**Issue by tbreloff** [JuliaPlots/PlotRecipes.jl#2]
Finish up spectral graphs, then tackle tree layouts and directed graphs.

See also: https://github.com/JuliaPlots/RecipesBase.jl/issues/3

**Comment by tbreloff** [comment]
Decent progress:

```julia
graphplot(A, markersize=20rand(n)+10, marker_z=rand(n), lc=:blues, dim=3)
```

![tmp](https://cloud.githubusercontent.com/assets/933338/16069720/a880f474-329e-11e6-9ae4-c1b9e2ffbc55.png)

**Comment by tbreloff** [comment]
![tmp](https://cloud.githubusercontent.com/assets/933338/16093627/9da7b26a-330a-11e6-9733-9d28d5bab604.png)

![tmp](https://cloud.githubusercontent.com/assets/933338/16094180/0dd2edf0-330d-11e6-8596-d12b0b8d5393.png)

**Comment by Evizero** [comment]
nice

**Comment by tbreloff** [comment]
Thinking about tree-based layouts for directed graphs...  I'd really prefer that GraphLayouts.jl was the source for layout algorithms, but I worry about the speed at which the dependencies will go away (i.e. Compose and JuMP)

In the meantime, I'd like to get **something** working in PlotRecipes.  Here are some ideas:

- https://www.youtube.com/watch?v=N0rWr8c1CeY
- http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.56.4906&rep=rep1&type=pdf
- https://cs.brown.edu/~rt/gdhandbook/chapters/hierarchical.pdf  (I think this is what is implemented in GraphLayout? probably too ambitious)

**Comment by tbreloff** [comment]
Here's another: http://web2-clone.research.att.com/export/sites/att_labs/groups/infovis/res/legacy_papers/DBLP-conf-gd-CarmelHK02.pdf

**Comment by tbreloff** [comment]
http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.437.3177&rep=rep1&type=pdf

**Comment by tbreloff** [comment]
http://ac.els-cdn.com/S0925772105000313/1-s2.0-S0925772105000313-main.pdf?_tid=d2da40a0-4396-11e6-a5b9-00000aacb360&acdnat=1467822754_24f87632c06119b159157d66890ddd7a

**Comment by tbreloff** [comment]
See section 2.3 of http://link.springer.com/chapter/10.1007%2F978-3-540-31843-9_25#page-1

**Comment by tbreloff** [comment]
This is roughly the "section 2.3" algorithm:

```julia
using PlotRecipes; pyplot(size=(300,300))
n = 40
s, d = Plots.unzip(unique([(rand(1:n),rand(1:n)) for i=1:60]))
s = convert(Vector{Int}, s)
d = convert(Vector{Int}, d)
w = ones(length(s));

basex, basey = rand(n), rand(n)
@gif for i=1:100
    x, y = PlotRecipes.by_axis_local_stress_graph(s,d,w, x=copy(basex),y=copy(basey),maxiter=i)
    graphplot(s,d,w, m=(linspace(20,40,n),:inferno), l=(3,:black), x=x, y=y, series_annotations=map(string,1:n), curves=false)
end
```

![tmp](https://cloud.githubusercontent.com/assets/933338/16698919/ee1f9e76-451e-11e6-8936-881551f120dd.gif)

**Comment by tbreloff** [comment]
Progress with trees:

```julia
using PlotRecipes; gr(size=(500,500))
s = [1,2,2,3,3,2,5,4,7]
d = [2,3,4,5,6,7,6,5,4]
w = ones(length(s))
n = max(maximum(s), maximum(d))

graphplot(s,d,w,func=:tree,series_annotations=map(string,1:n),root=:top)
```

![tmp](https://cloud.githubusercontent.com/assets/933338/16701854/93422160-4530-11e6-8118-11bbccb39a23.png)

**Comment by ChrisRackauckas** [comment]
What about a recipe for LightGraphs.jl? @sbromberger

**Comment by sbromberger** [comment]
Sure. What do we need to do?

**Comment by my-little-repository** [comment]
You should definitely put your last [example](https://github.com/JuliaPlots/PlotRecipes.jl/issues/2#issuecomment-231476190) on the main page of the website. I just spent one hour trying to figure out how to get these nice round labeled vertices and finally ended up here after browsing the documentation, examples and code source.

**Comment by mkborregaard** [comment]
Yeah, the readme examples on PlotRecipes are broken - I think you've already opened an issue there? Sorry for the confusion.
